jax/_src/numpy/lax_numpy.py

```diff
@@ -2271,8 +2271,15 @@ def identity(n: DimSize, dtype: Optional[DTypeLike] = None) -> Array:
 def arange(start: DimSize, stop: Optional[DimSize] = None,
            step: Optional[DimSize] = None, dtype: Optional[DTypeLike] = None) -> Array:
   dtypes.check_user_dtype_supported(dtype, "arange")
-  require = partial(core.concrete_or_error, None)
-  msg = "It arose in jax.numpy.arange argument `{}`.".format
+  util._check_arraylike_or_none("arange", start, stop, step)
+  msg = "It arose in jax.numpy.arange argument `{}`."
```

Comment:
The design here concerns me a little bit, because we're performing the string interpolation for the error message irrespective of whether the error actually happens.

This has caused performance problems in the past, especially for logging.

I admit it is not new in this change.

Reply:
Replaced the string interpolation with string constants.